src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs

```diff
@@ -1312,5 +1312,166 @@ public void Match_Wordchar()
             }
             Assert.Empty(ambiguous);
         }
+
+        [Theory]
+        [InlineData(RegexOptions.None)]
+        [InlineData(RegexOptions.Compiled)]
+        [InlineData(RegexHelpers.RegexOptionNonBacktracking)]
+        public void StressTestDeepNestingOfConcat(RegexOptions options)
+        {
+            if (PlatformDetection.IsNetFramework && options == RegexHelpers.RegexOptionNonBacktracking)
+            {
+                return;
+            }
+
+            string pattern = string.Concat(Enumerable.Repeat("([a-z]", 1000).Concat(Enumerable.Repeat(")", 1000)));
+            string input = string.Concat(Enumerable.Repeat("abcde", 200));
```

Comment:
Nit: could you lift these constant values out into named const locals?  I'm also wondering if 1000 is sufficient, especially on Linux where stacks are typically much larger.

Reply:
I introduced `RegexHelpers.StressTestNestingDepth` and set it  to 10000. I'm getting 
`The program '[55724] dotnet.exe' has exited with code 3221225477 (0xc0000005) 'Access violation'.`
so need to investigate this further.

Reply:
This happens only in `NonBacktracking` mode, the other cases seem fine.

Reply:
I'm investigating this next, but will set it back to 1000 for now to finish these edits first,